lib/matplotlib/artist.py

```diff
@@ -1417,6 +1418,80 @@ def set_mouseover(self, mouseover):
     mouseover = property(get_mouseover, set_mouseover)  # backcompat.
 
 
+class ColorizingArtist(Artist):
+    def __init__(self, norm=None, cmap=None):
+        """
+        Parameters
+        ----------
+        norm : `colors.Normalize` (or subclass thereof) or str or `cm.Colorizer` or None
+            The normalizing object which scales data, typically into the
+            interval ``[0, 1]``.
+            If a `str`, a `colors.Normalize` subclass is dynamically generated based
+            on the scale with the corresponding name.
+            If `cm.Colorizer`, the norm an colormap on the `cm.Colorizer` will be used
+            If *None*, *norm* defaults to a *colors.Normalize* object which
+            initializes its scaling based on the first data processed.
+        cmap : str or `~matplotlib.colors.Colormap`
+            The colormap used to map normalized data values to RGBA colors.
+        """
+
+        Artist.__init__(self)
+
+        self._A = None
+        if isinstance(norm, Colorizer):
+            self.colorizer = norm
+            if cmap:
+                raise ValueError("Providing a `cm.Colorizer` as the norm while "
+                                 "at the same time as a `cmap` is not supported..")
+        else:
+            self.colorizer = Colorizer(cmap, norm)
+
+        self._id_colorizer = self.colorizer.callbacks.connect('changed', self.changed)
+        self.callbacks = cbook.CallbackRegistry(signals=["changed"])
+
+    def set_array(self, A):
+        """
+        Set the value array from array-like *A*.
+
+        Parameters
+        ----------
+        A : array-like or None
+            The values that are mapped to colors.
+
+            The base class `.VectorMappable` does not make any assumptions on
+            the dimensionality and shape of the value array *A*.
+        """
+        if A is None:
+            self._A = None
+            return
+
+        A = cbook.safe_masked_invalid(A, copy=True)
+        if not np.can_cast(A.dtype, float, "same_kind"):
+            raise TypeError(f"Image data of dtype {A.dtype} cannot be "
+                            "converted to float")
+
+        self._A = A
+        if not self.norm.scaled():
```

Comment:
Side-note: We may have to reconsider how auto-scaling is executed. Currently, the first array to be set triggers auto-scaling. This does not work well when multiple ColorizingArtists share a Colorizer and thus a Norm. It may be reasonable to defer scaling to the first time that the norm (or the colorizing pipeline) is executed and collect the data limits from all participating Artists at that time. However, this requires a connection from the Colorizer to all associated artists. - I suppose this should be handled on the Colorizer level, The norm itself should likely stay Artist-agnostic.

This should be considered / changed as part of the Artist connection refactoring (Note 3.1)

Reply:
I added this to the list above

Reply:
Marking as resolved because this is tracked above.